inspire/modules/formatter/views.py

```diff
@@ -84,3 +89,57 @@ def get_latex_file(recid, latex_format):
             )
         }
     )
+
+
+@blueprint.route("/export-as/<path:export_format>/<path:ids>",
+                 methods=['GET', 'POST'])
+def download_all_latex(ids, export_format):
+    """
+    Export file according to format
+    :param ids: the ids of the records.
+    :param export_format: the type of the file the user needs to download.
+    :return: the downloaded file.
+    """
+    out = []
+    ids_list = ids.split(',')
+    suffix = ''
+    for element in ids_list:
+        record = get_record(element)
+        if export_format == 'bibtex':
+            results = Bibtex(record).format() + '\n'
+            suffix = 'bibtex'
+        elif export_format in ('latex_eu', 'latex_us'):
+            results = Latex(record, export_format).format() + '\n'
+            suffix = 'tex'
+        elif export_format == 'cv_latex':
+            results = Cv_latex(record).format() + '\n'
+        elif export_format in ('cv_latex_html', 'cv_latex_text'):
+            results = Cv_latex_html_text(record, export_format).format() + '\n'
+
+        generator = (cell for row in results
+                     for cell in row)
+        out.append(generator)
+    output = itertools.chain()
+    for gen in out:
+        output = itertools.chain(output, gen)
+
+    return Response(
+        output,
+        mimetype="text/plain",
+        headers={
+            "Content-Disposition": "attachment;filename=export_as_{0}.{1}".format(
+                export_format, suffix
+            )}
+    )
+
+
+@blueprint.route("/get-results/<path:params>",
+                 methods=['GET', ])
+@blueprint.route("/get-results/", defaults={'params': ''},
+                 methods=['GET', ])
+def get_results(params):
```

Comment:
Yes, they are different. If we find the problem then we can use this search
function.

2015-10-23 13:43 GMT+02:00 Javier Martin Montull <notifications@github.com>:

> In inspire/modules/formatter/views.py
> <https://github.com/inspirehep/inspire-next/pull/427#discussion_r42856180>
> :
>
> > +
> > +    return Response(
> > +        output,
> > +        mimetype="text/plain",
> > +        headers={
> > +            "Content-Disposition": "attachment;filename=export_as_{0}.{1}".format(
> > +                export_format, suffix
> > +            )}
> > +    )
> > +
> > +
> > +@blueprint.route("/get-results/<path:params>",
> > +                 methods=['GET', ])
> > +@blueprint.route("/get-results/", defaults={'params': ''},
> > +                 methods=['GET', ])
> > +def get_results(params):
>
> Are the recids and the control_number different?? If that is the case we
> should check why is that.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/inspirehep/inspire-next/pull/427/files#r42856180>.
>